samtranslator/translator/translator.py

```diff
@@ -268,6 +278,103 @@ def _get_resources_to_iterate(
+                self.document_errors.append(e)
+
+            for connector_logical_id, connector_dict in resource["Connectors"].items():
+                try:
+                    connector_logical_id = source_logical_id + connector_logical_id
+                    # can't use sam_expect since this is neither a property nor a resource attribute
+                    if not isinstance(connector_dict, dict):
+                        raise InvalidResourceException(
+                            connector_logical_id, f"{source_logical_id}.{connector_logical_id} should be a map."
+                        )
+
+                    generated_connector = self._get_generated_connector(
+                        source_logical_id,
+                        connector_logical_id,
+                        connector_dict,
+                    )
+
+                    if not verify_unique_logical_id(generated_connector, resources):
+                        raise DuplicateLogicalIdException(
+                            source_logical_id, connector_logical_id, generated_connector.resource_type
+                        )
+                    connectors.append(generated_connector)
+                except (InvalidResourceException, DuplicateLogicalIdException) as e:
+                    self.document_errors.append(e)
+
+        return connectors
+
+    def _get_generated_connector(
+        self, source_logical_id: str, connector_logical_id: str, connector_dict: Dict[str, Any]
+    ) -> Resource:
+        """
+        Generates the connector resource from the embedded connector
+
+        :param str source_logical_id: Logical id of the resource the connector is attached to
+        :param str connector_logical_id: Logical id of the connector
+        :param dict connector_dict: The properties of the connector including the Destination, Permissions and optionally the SourceReference
+        :return: The generated SAMConnector resource
+        """
+        connector = copy.deepcopy(connector_dict)
+        connector["Type"] = SamConnector.resource_type
+
+        # No need to raise an error for this instance as the error will be caught by the parser
+        if "Properties" in connector_dict and isinstance(connector_dict["Properties"], dict):
+            properties = connector["Properties"]
+            properties["Source"] = {"Id": source_logical_id}
+            if "SourceReference" in properties:
+                sam_expect(
+                    properties.get("SourceReference"),
+                    connector_logical_id,
+                    f"{connector_logical_id}.Properties.SourceReference",
+                ).to_be_a_map()
+
+                # can't allow user to override the Id using SourceReference
+                if "Id" in properties["SourceReference"]:
+                    raise InvalidResourceException(connector_logical_id, "Id shouldn't be defined in 'SourceReference'")
+
+                properties["Source"].update(properties["SourceReference"])
```

Comment:
I thought we only allow `Qualifier`? None of the other properties should work alongside `Id` AFAIK.

Reply:
If they use anything other than `Qualifier` then it'll throw an error from the connectors class https://github.com/aws/serverless-application-model/blob/b4f9ea1b083469a304ae248e26f3b2afb3a8b6f7/samtranslator/model/connector/connector.py#L95